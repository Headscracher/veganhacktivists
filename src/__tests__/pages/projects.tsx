import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import Projects from '../../pages/projects';

const image = {
  fields: { file: { url: null, details: { image: { width: 0, height: 0 } } } },
};

const projects = [
  {
    fields: {
      name: 'Activist Hub',
      image,
      date: '2021-04-01',
      year: 2021,
      team: { fields: { name: 'Test Team', color: 'black', icon: '' } },
      // description: (
      //   <p>
      //     The days of wondering whether or not our activism is effective are
      //     over. Activist Hub is the world&apos;s first street outreach dashboard
      //     that provides volunteers with the ability to monitor and understand
      //     their effectiveness through real data and analytics (DnA). Personal
      //     and group results will help everyone interested in making informed and
      //     strategic decisions for the animals!
      //   </p>
      // ),
      url: 'https://activisthub.org',
    },
  },
  {
    fields: {
      name: 'Sehati Sanctuary',
      image,
      date: '2021-03-01',
      year: 2021,
      team: { fields: { name: 'Test Team', color: 'black', icon: '' } },
      // description: (
      //   <p>
      //     Sehati Animal Sanctuary is the first farmed animal sanctuary in
      //     Indonesia which provides a loving haven to animals rescued from
      //     needless exploitation, violence, and slaughter. The sanctuary supports
      //     human-animal bonding and serves to educate the public about animal
      //     rights in order to reduce worldwide animal suffering.
      //   </p>
      // ),
      url: 'https://sehatisanctuary.org',
    },
  },
];

it('should render correctly', () => {
  const { asFragment } = render(<Projects projects={projects as any} />);
  expect(asFragment()).toMatchSnapshot();
});

const expectedProjectsPerYear = [
  {
    year: 2018,
    projects: [],
  },
  {
    year: 2019,
    projects: [],
  },
  {
    year: 2020,
    projects: [],
  },
  {
    year: 2021,
    projects: ['Activist Hub', 'Sehati Sanctuary'],
  },
];

expectedProjectsPerYear.forEach(({ year, projects: expected }) => {
  it(`should show only projects made in ${year}`, () => {
    render(<Projects projects={projects as any} />);
    fireEvent.click(screen.getByText(year));
    const actual = Array.from(
      document.querySelectorAll('h1.text-4xl.font-bold')
    ).map(({ innerHTML }) => innerHTML);
    expect(actual).toEqual(expected);
  });
});

it('should show all projects', () => {
  const expected = expectedProjectsPerYear.reduce(
    (accumulator, { projects }) => [...accumulator, ...projects],
    [] as string[]
  );
  render(<Projects projects={projects as any} />);
  fireEvent.click(screen.getByText('View all'));
  const actual = Array.from(
    document.querySelectorAll('h1.text-4xl.font-bold')
  ).map(({ innerHTML }) => innerHTML);
  expect(actual).toEqual(expected);
});
