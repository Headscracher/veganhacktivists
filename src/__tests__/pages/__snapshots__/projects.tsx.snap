// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<DocumentFragment>
  <div>
    <div
      class="relative flex justify-start bg-center"
    >
      <div
        style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
      >
        <img
          alt=""
          data-nimg="true"
          decoding="async"
          sizes="100vw"
          src="/_next/image?url=%2Ffoo.png&w=3840&q=75"
          srcset="/_next/image?url=%2Ffoo.png&w=640&q=75 640w, /_next/image?url=%2Ffoo.png&w=750&q=75 750w, /_next/image?url=%2Ffoo.png&w=828&q=75 828w, /_next/image?url=%2Ffoo.png&w=1080&q=75 1080w, /_next/image?url=%2Ffoo.png&w=1200&q=75 1200w, /_next/image?url=%2Ffoo.png&w=1920&q=75 1920w, /_next/image?url=%2Ffoo.png&w=2048&q=75 2048w, /_next/image?url=%2Ffoo.png&w=3840&q=75 3840w"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; object-position: center;"
        />
      </div>
      <div
        class="flex flex-col justify-center w-1/2 z-10"
      >
        <div
          class="py-20 md:py-20 lg:py-20"
        >
          <div
            style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
          >
            <div
              style="box-sizing: border-box; display: block; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMzM5LjU5OTk5OTk5OTk5OTk3IiBoZWlnaHQ9IjMzOS41OTk5OTk5OTk5OTk5NyIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
                style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
              />
            </div>
            <img
              alt="Building projects with impact"
              data-nimg="true"
              decoding="async"
              src="/_next/image?url=%2Ffoo.png&w=750&q=75"
              srcset="/_next/image?url=%2Ffoo.png&w=384&q=75 1x, /_next/image?url=%2Ffoo.png&w=750&q=75 2x"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
          </div>
        </div>
      </div>
      <div
        class="absolute inset-0 overflow-hidden"
      >
        <svg
          class="text-white overflow-visible absolute top-0 right-0 transform translate-x-1/4 -translate-y-2/4 lg:-translate-y-3/4
    "
          height="60vw"
          width="60vw"
        >
          <circle
            class="stroke-current"
            cx="30vw"
            cy="30vw"
            fill-opacity="0"
            opacity="0.1"
            r="30vw"
            stroke-width="20px"
          />
        </svg>
        <svg
          class="text-white overflow-visible absolute bottom-0 left-0 transform -translate-x-1/4 translate-y-2/4 lg:translate-y-3/4
    "
          height="68vw"
          width="68vw"
        >
          <circle
            class="stroke-current"
            cx="34vw"
            cy="34vw"
            fill-opacity="0"
            opacity="0.2"
            r="34vw"
            stroke-width="20px"
          />
        </svg>
      </div>
    </div>
    <div
      class="relative overflow-visible hidden md:block"
    >
      <div
        class="absolute left-0 bottom-0 bg-white w-16 h-16 opacity-100"
      />
      <div
        class="absolute left-0 top-0 bg-orange w-16 h-16 opacity-100"
      />
      <div
        class="absolute left-16 bottom-0 bg-magenta w-32 h-32 opacity-100"
      />
      <div
        class="absolute right-0 -top-16 bg-yellow w-32 h-32 opacity-100"
      />
      <div
        class="absolute right-32 bottom-16 bg-yellow_orange w-16 h-16 opacity-100"
      />
      <div
        class="absolute right-32 bottom-0 bg-white w-16 h-16 opacity-100"
      />
    </div>
    <div
      class="text-grey content-center mx-auto my-12 md:w-1/2 drop-shadow-2xl text-2xl"
    >
      <h1
        class="mb-8"
      >
        <span
          class="font-italic text-3xl"
        >
          Our
        </span>
         
        <span
          class="font-mono font-bold text-5xl uppercase text-black"
        >
          projects
        </span>
      </h1>
      <p>
        We're constantly working on new projects every month, whether they're ideas of our own or supporting organizations and activists that reach out to us. Below is a list of our work, and which of our teams worked on it.
      </p>
    </div>
    <div
      class="flex flex-col space-y-20 items-center mx-auto drop-shadow-2xl text-2xl pb-20"
    >
      <div
        class="flex flex-wrap place-content-center justify-center"
      >
        <button
          type="button"
        >
          <div
            class="border-2 border-gray-200 border-opacity-50 p-3 bg-gray text-white w-40 uppercase flex-1 m-1"
          >
            View all
          </div>
        </button>
        <button
          type="button"
        >
          <div
            class="border-2 border-gray-200 border-opacity-50 p-3 w-40 flex-1 m-1"
          >
            2021
          </div>
        </button>
        <button
          type="button"
        >
          <div
            class="border-2 border-gray-200 border-opacity-50 p-3 w-40 flex-1 m-1"
          >
            2020
          </div>
        </button>
        <button
          type="button"
        >
          <div
            class="border-2 border-gray-200 border-opacity-50 p-3 w-40 flex-1 m-1"
          >
            2019
          </div>
        </button>
        <button
          type="button"
        >
          <div
            class="border-2 border-gray-200 border-opacity-50 p-3 w-40 flex-1 m-1"
          >
            2018
          </div>
        </button>
      </div>
      <div
        class="w-3/4"
      >
        <div
          class="flex flex-col sm:flex-row justify-between"
        >
          <div
            class="flex-shrink"
          >
            <div
              style="overflow: hidden; box-sizing: border-box; display: inline-block; position: relative; width: 300px; height: 300px;"
            >
              <img
                alt="Activist Hub"
                data-nimg="true"
                decoding="async"
                src="/_next/image?url=https%3Anull&w=640&q=75"
                srcset="/_next/image?url=https%3Anull&w=384&q=75 1x, /_next/image?url=https%3Anull&w=640&q=75 2x"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
            </div>
          </div>
          <div
            class="sm:pl-10 col-span-2 text-left"
          >
            <h1
              class="text-4xl font-bold mb-5"
            >
              Activist Hub
            </h1>
            <div
              class="text-xl"
            />
            <div
              class="mt-10 flex flex-wrap items-center"
            >
              <a
                href="https://activisthub.org"
                rel="noreferrer"
                target="_blank"
              >
                <div
                  class="p-3 px-4 py-2 text-2xl border-l-8 bg-w-x2 ease-out duration-1000 cursor-pointer hover:shadow-fill-green bg-grey-dark border-green text-white font-mono"
                >
                  activisthub.org
                </div>
              </a>
              <span
                class="font-bold sm:pl-5"
              >
                <span
                  class="text-grey"
                >
                  Apr 2021
                </span>
                 - 
                <span
                  style="color: black;"
                >
                  Test Team
                </span>
              </span>
            </div>
          </div>
        </div>
        <div
          class="flex flex-col sm:flex-row justify-between mt-10"
        >
          <div
            class="flex-shrink"
          >
            <div
              style="overflow: hidden; box-sizing: border-box; display: inline-block; position: relative; width: 300px; height: 300px;"
            >
              <img
                alt="Sehati Sanctuary"
                data-nimg="true"
                decoding="async"
                src="/_next/image?url=https%3Anull&w=640&q=75"
                srcset="/_next/image?url=https%3Anull&w=384&q=75 1x, /_next/image?url=https%3Anull&w=640&q=75 2x"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
            </div>
          </div>
          <div
            class="sm:pl-10 col-span-2 text-left"
          >
            <h1
              class="text-4xl font-bold mb-5"
            >
              Sehati Sanctuary
            </h1>
            <div
              class="text-xl"
            />
            <div
              class="mt-10 flex flex-wrap items-center"
            >
              <a
                href="https://sehatisanctuary.org"
                rel="noreferrer"
                target="_blank"
              >
                <div
                  class="p-3 px-4 py-2 text-2xl border-l-8 bg-w-x2 ease-out duration-1000 cursor-pointer hover:shadow-fill-green bg-grey-dark border-green text-white font-mono"
                >
                  sehatisanctuary.org
                </div>
              </a>
              <span
                class="font-bold sm:pl-5"
              >
                <span
                  class="text-grey"
                >
                  Mar 2021
                </span>
                 - 
                <span
                  style="color: black;"
                >
                  Test Team
                </span>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="relative overflow-visible hidden md:block"
    >
      <div
        class="absolute left-0 top-0 bg-white w-16 h-16 opacity-100"
      />
      <div
        class="absolute left-0 bottom-0 bg-gray-lighter w-16 h-16 opacity-100"
      />
      <div
        class="absolute right-0 bottom-0 bg-gray-lighter w-16 h-16 opacity-100"
      />
    </div>
    <div
      class="bg-gray md:p-36 flex flex-row justify-center"
    >
      <div
        class="bg-green flex p-2 md:p-8"
      >
        <div
          style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
        >
          <div
            style="box-sizing: border-box; display: block; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMzAwIiBoZWlnaHQ9IjMwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
              style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
            />
          </div>
          <noscript />
          <img
            alt="Lamp"
            data-nimg="true"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
          />
        </div>
      </div>
      <div
        class="bg-gray-background text-left p-3 md:p-10"
      >
        <h4
          class="text-2xl mb-3 md:mb-8"
        >
          Have an idea for a project?
        </h4>
        <p
          class="mb-3 md:mb-8"
        >
          <strong
            class="font-bold"
          >
            We're all ears
          </strong>
          ! Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua
        </p>
        <a
          href="/projects"
        >
          <div
            class="p-3 px-4 py-2 text-2xl border-l-8 bg-w-x2 ease-out duration-1000 cursor-pointer hover:shadow-fill-green text-grey-dark border-green bg-w-x2 bg-white w-72"
          >
            Suggest a project idea
          </div>
        </a>
      </div>
    </div>
    <div
      class="bg-black"
    >
      <div
        class="content-center mx-auto md:w-1/2 drop-shadow-2xl text-2xl pb-12"
      >
        <p
          class="mb-12 text-grey-dark pt-16"
        >
          <span
            class="text-5xl font-mono text-white"
          >
            JOIN OUR TEAM
          </span>
        </p>
        <div
          style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
        >
          <div
            style="box-sizing: border-box; display: block; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNTAwIiBoZWlnaHQ9IjUwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
              style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
            />
          </div>
          <noscript />
          <img
            alt="Compassion, Creativity, Code"
            data-nimg="true"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
        </div>
        <p
          class="pb-5 mt-16 text-white"
        >
          Are you a developer, designer, writer, or other creative professional wanting to use skills for the animals? We're always looking for more vegan activist volunteers to join our team! If you're interested and have the time to spare, click to see our positions.
        </p>
        <div
          class="flex justify-center flex-wrap"
        >
          <a
            href="/people/team"
          >
            <div
              class="p-3 px-4 py-2 text-2xl border-l-8 bg-w-x2 ease-out duration-1000 cursor-pointer hover:shadow-fill-green text-grey-dark border-green bg-w-x2 bg-white m-5 font-mono text-sm"
            >
              Meet the Team
            </div>
          </a>
          <a
            href="/join"
          >
            <div
              class="p-3 px-4 py-2 text-2xl border-l-8 bg-w-x2 ease-out duration-1000 cursor-pointer hover:shadow-fill-green text-grey-dark border-green bg-w-x2 bg-white m-5 font-mono text-sm"
            >
              Apply to Join
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
